### PR TITLE
[Example] Collaborative Block Text Editor

### PR DESCRIPTION
**Pull request by stevenfabre**
Adding a new example to demo how Liveblocks can be used to create a collaborative block-based text editor similar to products like [Notion](http://notion.so) and [Clover](http://cloverapp.com).

block-text-editor-advanced

### Slate integration

- [x]  ⚠️  Synchronize slate with Liveblocks properly
- [x]  Code cleanup
- [x]  Small CSS fixes
- [x]  Drag and drop animations
- [x] Toolbar positioning
- [x]  Comments / initial state with explanation

### Nice to have if done well

- [ ]  Undo redo
- [ ]  Multiplayer selection at character level

### Done

- [x]  Hide toolbar on the document title
- [x]  Fix forced layout on the second block
- [x]  Disable document title drag and drop
- [x]  Remove recursive id generation (withNodeId)
- [x]  Placeholder design
- [x]  Video block
- [x]  Codesandbox block
- [x] Insert paragraph after a line break

### CODE REVIEW

**Comment by GuillaumeSalles** [comment]
Amazing work! 👏 

I found a few issues:
- Offset is off when splitting a block after formatted text
- Merging block erase formatting

See video for the two issues 
https://user-images.githubusercontent.com/7273466/180910104-5fff49ff-7335-4785-bc6e-bb7becef9987.mov

Did you consider using a simple loop to go over each character to compute the offset in `getHtmlIndexPositionFromInnerTextPosition` instead of a regex?

File: examples/nextjs-block-text-editor-advanced/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+<img src="https://raw.githubusercontent.com/liveblocks/liveblocks/main/.github/assets/examples/block-text-editor.png" width="500" alt="Collaborative Block Text Editor" />
+
+## Getting started
+
+- Install all dependencies with `npm install`
+- Create an account on [liveblocks.io](https://liveblocks.io/dashboard)
+- Copy your **public** key from the
+  [dashboard](https://liveblocks.io/dashboard/apikeys)
+- Create an `.env.local` file and add your **public** key as the
+  `NEXT_PUBLIC_LIVEBLOCKS_PUBLIC_KEY` environment variable
```
Comment (FlowFlorent) [inline examples/nextjs-block-text-editor-advanced/README.md:32]:
```suggestion
  `LIVEBLOCKS_SECRET_KEY` environment variable
```

File: examples/nextjs-block-text-editor-advanced/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+- Run `npm run dev` and go to [http://localhost:3000](http://localhost:3000)
+
+### CodeSandbox
+
+After forking
+[this example](https://codesandbox.io/s/github/liveblocks/liveblocks/tree/main/examples/block-text-editor)
+on CodeSandbox, create the `NEXT_PUBLIC_LIVEBLOCKS_PUBLIC_KEY` environment
```
Comment (FlowFlorent) [inline examples/nextjs-block-text-editor-advanced/README.md:39]:
```suggestion
on CodeSandbox, create the `LIVEBLOCKS_SECRET_KEY` environment
```

**Comment by FlowFlorent** [comment]
Found a tiny bug:
When focusing the title, the user's focus on the text is not cleared

https://user-images.githubusercontent.com/6947265/181034782-013e8289-e000-4ab6-b2cc-f5ceae6eabe4.mp4

File: examples/nextjs-block-text-editor-advanced/src/components/BlockText copy.tsx [hunk 3]
```diff
@@ -0,0 +1,208 @@
+import styles from "../../styles/BlockText.module.css";
```
Comment (GuillaumeSalles) [inline examples/nextjs-block-text-editor-advanced/src/components/BlockText copy.tsx:1]: To delete 👀

Comment (stevenfabre) [reply examples/nextjs-block-text-editor-advanced/src/components/BlockText copy.tsx:1]: Oops my bad. Will delete!

**Pull request merged**